From f6bc225c15dea787aceaa30d207a0cb90415ec5e Mon Sep 17 00:00:00 2001
From: Locus Wei-Han Chen <locus84@andestech.com>
Date: Wed, 9 Oct 2024 13:09:11 +0800
Subject: [PATCH 125/168] riscv: dts: andes: add dmas property for mmc node
 (#210)

As MMC does not include the DMA property, the SD card cannot be used.
We add the DMA property for mmc node in the ax25mp_amp_c2_64_d_ae350.dts.

[Bugzill 31844](http://e-andes.andestech.com/bugzilla5/show_bug.cgi?id=31944)

Signed-off-by: Locus Wei-Han Chen <locus84@andestech.com>
Reviewed-on: https://gitea.andestech.com/RD-SW/linux/pulls/210
Reviewed-by: Leo Yu-Chi Liang <ycliang@andestech.com>
Reviewed-by: Charles Ci-Jyun Wu <dminus@andestech.com>
Co-authored-by: Locus Wei-Han Chen <locus84@andestech.com>
Co-committed-by: Locus Wei-Han Chen <locus84@andestech.com>
---
 .../boot/dts/andes/ax25mp_amp_c2_64_d_ae350.dts | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/arch/riscv/boot/dts/andes/ax25mp_amp_c2_64_d_ae350.dts b/arch/riscv/boot/dts/andes/ax25mp_amp_c2_64_d_ae350.dts
index 03e1de946108..c6a7c34e6350 100644
--- a/arch/riscv/boot/dts/andes/ax25mp_amp_c2_64_d_ae350.dts
+++ b/arch/riscv/boot/dts/andes/ax25mp_amp_c2_64_d_ae350.dts
@@ -237,21 +237,26 @@ mac@e0100000 {
 
 	mmc@f0e00000 {
 		compatible = "andestech,atfsdc010";
-		max-frequency = <0x5f5e100>;
+		reg = <0x00 0xf0e00000 0x00 0x1000>;
+		interrupts = <0x12 0x04>;
+		interrupt-parent = <0x04>;
 		clock-freq-min-max = <0x61a80 0x5f5e100>;
+		max-frequency = <0x5f5e100>;
 		fifo-depth = <0x10>;
-		reg = <0x0 0xf0e00000 0x0 0x1000>;
-		interrupts = <0x12 0x4>;
-		cap-sd-highspeed;
-		interrupt-parent = <0x4>;
+		dma-names = "rxtx";
+		dmas = <0x07 0x09>;
+		dma-coherent;
 	};
 
 	dma@f0c00000 {
 		compatible = "andestech,atcdmac300";
 		reg = <0x0 0xf0c00000 0x0 0x1000>;
-		interrupts = <0xa 0x4 0x40 0x4 0x41 0x4 0x42 0x4 0x43 0x4 0x44 0x4 0x45 0x4 0x46 0x4 0x47 0x4>;
+		interrupts = <0xa 0x4>;
 		dma-channels = <0x8>;
 		interrupt-parent = <0x4>;
+		#dma-cells = <0x01>;
+		dma-coherent;
+		phandle = <0x07>;
 	};
 
 	lcd@e0200000 {
-- 
2.34.1

